core/controllers/profile.py

```diff
@@ -311,6 +347,36 @@ def get(self):
 class SignupHandler(base.BaseHandler):
     """Provides data for the editor prerequisites page."""
 
+
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {},
+        'POST': {
+            'username': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            },
```

Comment:
Why not username validator here?

Reply:
I have added the validator,  please have a look.